src/server/network/network.ts

```diff
@@ -0,0 +1,152 @@
+import dgram from 'dgram';
+import dns from 'dns';
+import ip6addr from 'ip6addr';
+
+export type ClientListener = (event: string, ...args: any[]) => void;
+export type MessageCallback = (
+  team: TeamData,
+  payload: Buffer,
+  rinfo: dgram.RemoteInfo,
+) => void;
+
+export type TeamData = {
+  name: string;
+  address: string;
+  port: number;
+};
+
+export class UDPServers {
+  private servers: Array<UDPServer>;
+  private clients: Map<string, ClientListener>;
+
+  constructor(teams: Array<TeamData>) {
+    this.clients = new Map<string, ClientListener>();
+    this.servers = new Array<UDPServer>();
+
+    teams.forEach((team: TeamData) => {
+      // Determine if we are listening on a IPv4 or IPv6 address and resolve any DNS address
+      // If the specified address can be resolved to an IPv4 address prefer that over an IPv6 address
+      dns.lookup(
+        team.address,
+        { family: 0, all: true, hints: dns.ADDRCONFIG | dns.V4MAPPED },
+        (
+          error: NodeJS.ErrnoException | null,
+          addresses: dns.LookupAddress[],
+        ) => {
+          if (error) {
+            throw `UDPServer encountered an error resolving a DNS address: Error: ${error?.code}\n${error?.stack}`;
```

Comment:
Optional chaining shouldn't be needed, as `error` is guaranteed to be defined with the `if (error) {` guard.

```suggestion
            throw `UDPServer encountered an error resolving a DNS address: Error: ${error.code}\n${error.stack}`;
```